src/System.Linq.Expressions/tests/Ternary/TernaryTests.cs

```diff
@@ -1125,7 +1122,7 @@ private static void VerifyGenericWithClassRestriction<Tc>(bool condition, Tc a,
                     Enumerable.Empty<ParameterExpression>());
             Func<Tc> f = e.Compile(useInterpreter);
 
-            Assert.Same(condition ? a : b, f());
```

Comment:
I will stop commenting on these.  Other than to say, if Same was wrong, then these tests would be failing.  The fact that they're not failing means that we are actually getting the same object reference.